### PR TITLE
HDDS-2787. Use try-with-resources or close resource in finally block

### PR DESCRIPTION
**Pull request by vivekratnavel**
## What changes were proposed in this pull request?

Sonarlint fixes for BasicOzoneClientAdapterImpl, OzoneClientAdapterFactory, DatanodeChunkGenerator, RandomKeyGenerator

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2787

## How was this patch tested?

Verified that the project compiles successfully.

### CODE REVIEW

**Comment by vivekratnavel** [comment]
@elek @anuengineer @bharatviswa504 @adoroszlai @dineshchitlangia Please review

File: hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java [hunk 1]
```diff
@@ -203,10 +203,9 @@ protected void incrementCounter(Statistic objectsRead) {
   public OzoneFSOutputStream createFile(String key, boolean overWrite,
       boolean recursive) throws IOException {
     incrementCounter(Statistic.OBJECTS_CREATED);
-    try {
-      OzoneOutputStream ozoneOutputStream = bucket
-          .createFile(key, 0, replicationType, replicationFactor, overWrite,
-              recursive);
+    try (OzoneOutputStream ozoneOutputStream = bucket
+        .createFile(key, 0, replicationType, replicationFactor, overWrite,
+            recursive)) {
```
Comment (adoroszlai) [inline hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java]: This stream is returned to the caller for writing.  It cannot be closed in this method.

Comment (vivekratnavel) [reply hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java]: @adoroszlai Thanks for the review. You are right! I fixed this in the latest patch.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/DatanodeChunkGenerator.java [hunk 2]
```diff
@@ -95,45 +95,47 @@ public Void call() throws Exception {
...
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException(
+                "Pipeline ID is NOT defined, and no pipeline " +
+                    "has been found with factor=THREE"));
+        LOG.info("Using pipeline {}", pipeline.getId());
+      }
 
-    timer = getMetrics().timer("chunk-write");
+      xceiverClientSpi =
+          new XceiverClientManager(ozoneConf).acquireClient(pipeline);
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/DatanodeChunkGenerator.java]: Turns out that in addition to the client, the `XceiverClientManager` should be closed, too.

Comment (vivekratnavel) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/DatanodeChunkGenerator.java]: Good catch! Will fix it in the next patch!

**Pull request merged**